src/app/ReadClient.cpp

```diff
@@ -503,12 +525,6 @@ void ReadClient::OnPeerTypeChange(PeerType aType)
     mIsPeerLIT = (aType == PeerType::kLITICD);
 
     ChipLogProgress(DataManagement, "Peer is now %s LIT ICD.", mIsPeerLIT ? "a" : "not a");
-
-    // If the peer is no longer LIT, try to wake up the subscription and do resubscribe when necessary.
-    if (!mIsPeerLIT)
-    {
-        OnActiveModeNotification();
```

Comment:
So I am trying to understand why it's OK to remove this.  The reason this seems to be here is that it's envisioning us having two subscriptions, one of which (call it A) is watching for peer type change and one of which (call it B) is IsInactiveICDSubscription().  When the transition to SIT is observed, B will no longer get any OnActiveModeNotifications, so how will it get out of the "inactive ICD subscription" state?

Of course if A does not exist and we just have B, and it's in the "inactive ICD subscription" state and then the server changes to A SIT, I think we just get stuck; that seems to be a hole in the whole logic of this stuff, including the spec?

Reply:
The scenario is as below
Assume we have one inactive wildcard subscription between controller and ICD device. When ICD transitions from LIT to SIT, check-in message is sent to controller, and controller activate the sleepy subscription, right? then the operation mode attribute is changed, OnPeerTypeChange is called so that OnActiveModeNotification is called again and this subscription would be recreated.

Reply:
You are right, the current change does not work on the below scenario

> So I am trying to understand why it's OK to remove this. The reason this seems to be here is that it's envisioning us having two subscriptions, one of which (call it A) is watching for peer type change and one of which (call it B) is IsInactiveICDSubscription(). When the transition to SIT is observed, B will no longer get any OnActiveModeNotifications, so how will it get out of the "inactive ICD subscription" state?

Reply:
updated the code.
1. Update PeerTypeChange and call TriggerResubscriptionForLivenessTimeout when peer type is changed to LIT and state is inactive icd subscription, instead of  OnActiveModeNotification to help situation 1 in your question.
2. If icd transition from lit to sit, it would send check-in message, which would bring up the in active subscription.